File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/GliaSendMessagePreviewUseCase.java
package com.glia.widgets.chat.domain;

import com.glia.widgets.chat.data.GliaChatRepository;

public class GliaSendMessagePreviewUseCase {

    private final GliaChatRepository gliaChatRepository;

    public GliaSendMessagePreviewUseCase(GliaChatRepository gliaChatRepository) {
        this.gliaChatRepository = gliaChatRepository;
    }

    public void execute(String message) {
        gliaChatRepository.sendMessagePreview(message);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/GliaSendMessagePreviewUseCase.kt
package com.glia.widgets.chat.domain

import com.glia.widgets.chat.data.GliaChatRepository

internal class GliaSendMessagePreviewUseCase(private val gliaChatRepository: GliaChatRepository) {
    operator fun invoke(message: String?) {
        gliaChatRepository.sendMessagePreview(message)
    }
}
